### PR TITLE
Fixing cluster creation and cluster deletion fn

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
Fixes: #204
Signed-off-by: srijan-sivakumar <ssivakum@redhat.com>

### CODE REVIEW

**Comment by srijan-sivakumar** [comment]
one set of change done is that the way how the cluster is created is modified. Instead of the way mentioned and proposed in the doc, I'm destroying the clusters and then re-creating it. This seems to be the simplified and straightforward way rather than taking a round-about approach and handling multiple failure cases in the documented approach.

File: support/ops/gluster_ops/peer_ops.py [hunk 1]
```diff
@@ -155,6 +155,25 @@ def nodes_from_pool_list(self, node: str) -> list:
...
+        the try-except block.
+        If anyone were to come across a better syntax, do change this
+        monstrosity.
+        """
+        try:
+            if 'hostname' in pool_list_data.keys():
+                nodes.append(pool_list_data['hostname'])
+                return nodes
+        except:
+            pass
```
Comment (aujjwal-redhat) [inline support/ops/gluster_ops/peer_ops.py:176]: Should we just `pass` or add some message as well?

File: support/ops/gluster_ops/peer_ops.py [hunk 2]
```diff
@@ -184,49 +203,87 @@ def get_pool_list(self, node: str) -> list:
...
+                node_list.remove(value)
+                node_list.append(ip_val)
+        return node_list
+
+    def create_cluster(self, node_list: list) -> bool:
+        """
+        Creates a cluster out of given set of nodes irrespective
+        of their existing cluster configurations.
+        Args:
+            node_list (list): All nodes which are to be part of the cluster.
```
Comment (aujjwal-redhat) [inline support/ops/gluster_ops/peer_ops.py:248]: NIT: Return details in docstring missing.

File: tests/parent_test.py [hunk 3]
```diff
@@ -36,6 +36,7 @@ def __init__(self, mname: str, config_hashmap: dict, volume_type: str,
         self.client_list = list(client_details.keys())
 
         self.redant.start_glusterd()
+        self.redant.create_cluster(self.server_list)
```
Comment (aujjwal-redhat) [inline tests/parent_test.py:39]: I guess the function referring to is `_add_orphan_nodes_to_cluster` . It was replaced in the above lines.

File: support/ops/gluster_ops/peer_ops.py (hunk 2 above)
Comment (srijan-sivakumar) [reply support/ops/gluster_ops/peer_ops.py:248]: Ack. Will add that.

File: tests/parent_test.py (hunk 3 above)
Comment (srijan-sivakumar) [reply tests/parent_test.py:39]: Yes, the function hasn't been removed. Will do a cleanup.

File: support/ops/gluster_ops/peer_ops.py (hunk 1 above)
Comment (srijan-sivakumar) [reply support/ops/gluster_ops/peer_ops.py:176]: I think the comment detailed the reasoning behind the addition of an except clause. No need to log anything here.

Comment (aujjwal-redhat) [reply support/ops/gluster_ops/peer_ops.py:176]: Ack

**Review by aujjwal-redhat — APPROVED** [review]
LGTM

**Pull request merged**